src/main/java/io/cdap/plugin/gcp/gcs/actions/GCSDoneFileMarker.java

```diff
@@ -0,0 +1,169 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.gcp.gcs.actions;
+
+import com.google.auth.Credentials;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageException;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.cdap.etl.api.PipelineConfigurer;
+import io.cdap.cdap.etl.api.batch.BatchActionContext;
+import io.cdap.cdap.etl.api.batch.PostAction;
+import io.cdap.plugin.common.batch.action.Condition;
+import io.cdap.plugin.common.batch.action.ConditionConfig;
+import io.cdap.plugin.gcp.common.GCPConfig;
+import io.cdap.plugin.gcp.common.GCPUtils;
+import io.cdap.plugin.gcp.gcs.GCSPath;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * A post action plugin that creates a marker file with a given name in case of a succeeded, failed or completed
+ * pipeline.
+ */
+@Plugin(type = PostAction.PLUGIN_TYPE)
+@Name(GCSDoneFileMarker.NAME)
+@Description("Creates a marker file with a given name in case of a succeeded, failed or completed pipeline.")
+public class GCSDoneFileMarker extends PostAction {
+  public static final String NAME = "GCSDoneFileMarker";
+  public Config config;
+
+  /**
+   * Creates a file marker
+   *
+   * @param project
+   * @param path
+   * @param serviceAccount
+   * @param isServiceAccountFilePath
+   * @param context
+   * @throws IOException
+   */
+  private static void createFileMarker(String project, GCSPath path, String serviceAccount,
+                                       Boolean isServiceAccountFilePath,
+                                       BatchActionContext context) {
+
+    String cmekKey = context.getArguments().get(GCPUtils.CMEK_KEY);
+    Credentials credentials = null;
+    if (serviceAccount != null) {
+      try {
+        credentials = GCPUtils.loadServiceAccountCredentials(serviceAccount, isServiceAccountFilePath);
+      } catch (IOException e) {
+        throw new RuntimeException(String.format("Failed to load credentials from path %s.", serviceAccount), e);
+      }
+    }
+
+    Storage storage = GCPUtils.getStorage(project, credentials);
+
+    if (storage.get(path.getBucket()) == null) {
+      try {
+        GCPUtils.createBucket(storage, path.getBucket(), null, cmekKey);
```

Comment:
does this fail if the bucket is already present?

Reply:
If the bucket exists, the `createBucket` method won't get executed due to the condition statement in line 78.